### PR TITLE
feat(component-driver-mui-x-v9): add column reorder and row detail panel APIs

### PR DESCRIPTION
**Pull request by tangentlin**
## Summary

Added two new feature sets to `DataGridPremiumDriver`:

1. **Column reordering** via `reorderColumn(sourceField, targetField)` — drags a column header to a new position using HTML5 drag-and-drop. Validates that both columns are rendered, waits for the column's `aria-colindex` to change, and throws if the drop is not accepted (e.g., pinned or checkbox-selection columns).

2. **Row detail panel (master-detail)** via four new methods:
   - `isRowDetailExpanded(rowIndex)` — reads the toggle's `aria-expanded`
   - `expandRowDetail(rowIndex)` / `collapseRowDetail(rowIndex)` — click the toggle and wait for both the toggle state and panel DOM presence to settle (closes a race window between MUI's synchronous toggle update and async panel insertion/removal)
   - `getRowDetailPanel(rowIndex, driverClass?)` — returns a driver over the expanded panel content, or `null` if collapsed
   - `getRowDetailText(rowIndex)` — convenience wrapper returning the panel's text content or `null`

All methods include comprehensive error handling and detailed JSDoc explaining edge cases (e.g., jsdom's synthesized `dragover` coordinates, MUI's virtualizer rendering off-window rows without their panels).

Added full test coverage:
- DOM tests (`MasterDetailDataGridPremium.dom.test.ts`) validating expansion/collapse, panel access, error cases, and nonexistent rows
- E2E tests (`MasterDetailDataGridPremium.e2e.test.ts`) for real-browser behavior
- Interactive tests for `reorderColumn` in the existing `InteractiveDataGridPremium` suite
- Example UI and test fixtures demonstrating both features

## Checks

- [ ] `pnpm run check:type`
- [ ] `pnpm run check:lint`
- [ ] `pnpm run check:style`
- [ ] `pnpm test:dom` (component-driver-mui-x-v9-test)
- [ ] `pnpm test:e2e` (component-driver-mui-x-v9-test, all three browsers)

## Breaking change

- [ ] This PR introduces a breaking change

https://claude.ai/code/session_01AR5d7hzaovTA8zKPjSFdfy

### CODE REVIEW

File: packages/component-driver-mui-x-v9/src/components/datagrid/DataGridPremiumDriver.ts [hunk 1]
```diff
@@ -104,6 +109,13 @@ const headerCheckboxLocator = byCssSelector('[role=columnheader][data-field="__c
 const virtualScrollerLocator = byCssClass('MuiDataGrid-virtualScroller');
 const overlayWrapperLocator = byCssClass('MuiDataGrid-overlayWrapper');
 
+// Master-detail: the row's detail-panel toggle is a dedicated cell (MUI X's fixed internal
+// field name `__detail_panel_toggle__`) whose button carries its own `aria-expanded` — unlike
+// row grouping, where `aria-expanded` lives on the row itself. The expanded panel renders as
+// that row's next DOM sibling within the virtualized render zone (not nested inside the row),
+// so it is addressed with the adjacent-sibling combinator off the row rather than a descendant.
+const rowDetailToggleLocator = byCssSelector('.MuiDataGrid-detailPanelToggleCell');
```
[the remark this answers is not in the document]
Comment (tangentlin) [reply packages/component-driver-mui-x-v9/src/components/datagrid/DataGridPremiumDriver.ts:117]: Thanks for the look, but this one's not a bug — `.MuiDataGrid-detailPanelToggleCell` is the button itself, not a wrapping cell around it.

In `@mui/x-data-grid-pro`'s `GridDetailPanelToggleCell.js`, the class and `aria-expanded` are set on the exact same element:

```js
return jsx(rootProps.slots.baseIconButton, {
  className: classes.root, // → "MuiDataGrid-detailPanelToggleCell"
  'aria-expanded': isExpanded,
  ...
});
```

`baseIconButton` renders as a `<button>`. I confirmed this against the live rendered DOM too (jsdom probe) before implementing:

```html
<button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-detailPanelToggleCell ..."
        aria-expanded="false" aria-label="Expand">…</button>
```

The "Cell" in the class/component name is just MUI's internal naming (the component is literally called `GridDetailPanelToggleCell`), not an indication of a separate wrapping element the way `groupToggleLocator` needs `button` appended for the grouping toggle. Appending `button` here as suggested would actually break the locator, since there's no nested button to find.

---
_Generated by [Claude Code](https://claude.ai/code)_
